src/Lecture1.hs

```diff
@@ -98,7 +111,12 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+
+subString :: Int -> Int -> String -> String
+subString start end str = take (end - negToZero start + 1) (drop start str)
+    where 
+        negToZero :: Int -> Int
+        negToZero x = if x < 0 then 0 else x
```

Comment:
You can use standard function `max` here to write a bit shorter 🙂 

```suggestion
        negToZero x = max 0 x
```

Reply:
That's a nice one